05/DB.test.js

```diff
@@ -1 +1,234 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('Database', () => {
+    describe('insert()', () => {
+        it('Should resolve when item added to database',
+            async () => {
+                const db = new DB();
+                const newItem = { a: 1, b: 2 };
+                await expect(db.insert(newItem))
+                    .resolves.toBe(newItem);
+            });
+
+        it('Should return size of DB = 2 when two items inserted',
+            async () => {
+                const db = new DB();
+                await db.insert({ a: 1, b: 2 });
+                await db.insert({ a: 3, b: 4 });
+
+                const dbSize = db._rows.length;
+                expect(dbSize).toBe(2);
+            });
+
+        it('Should return id: 1 when first item inserted',
+            async () => {
+                const db = new DB();
+                const result = await db.insert({ a: 1, b: 2 });
+
+                expect(result.id).toBe(1);
+            });
+
+        it('Should reject when id is not a number',
+            async () => {
+                const db = new DB();
+                const itemData = { id: 'some string', a: 1, b: 2 };
+                await expect(db.insert(itemData))
+                    .rejects.toBe('ID can be only number!');
+            });
+
+        it('Should reject when id is duplicated',
+            async () => {
+                const db = new DB();
+                db._rows = [
+                    { a: 1, b: 2, id: 1 }
+                ];
+
+                const duplicatedItemID = { a: 3, b: 4, id: 1 };
+                await expect(db.insert(duplicatedItemID))
+                    .rejects.toBe('ID can\'t be duplicated!');
+            });
+    });
+    describe('select()', () => {
+        it('Should resolve when id is found',
+            async () => {
+                const db = new DB();
+                const itemData = await db.insert({ id: 2, a: 1, b: 2 });
+                await expect(db.select(2))
+                    .resolves.toBe(itemData);
+            });
+
+        it('Should reject when id is not found',
+            async () => {
+                const db = new DB();
+                db._rows = [
+                    { a: 1, b: 2, id: 1 }
+                ];
```

Comment:
Tutaj podobnie jak wcześniej. Bardzo możliwe, że np. `insert()` ma dodatkowe działania związane z optymalizacją zapytań. Dodanie od razu do `_rows` danych może powodować wolniejsze działanie (to może byc powód dla którego `_rows` jest "prywatne").

Reply:
Co ciekawe w mojej pierwszej wersji testów korzystałem w każdym teście z `insert()`. Później, jak spojrzałem na testy, coś mnie podkusiło, żeby to zmienić, bo uznałem, że będzie czytelniej – postaram się to zapamiętać haha 😀 .